docs/tutorial.rst

```diff
@@ -0,0 +1,912 @@
+  0.016362461737446838 sr
+  >>> print(hp.pixel_resolution)
+  439.74226071262797 arcmin
+
+  >>> coord = SkyCoord("00h42m44.3503s +41d16m08.634s", frame="galactic")
+  >>> print(hp.interpolate_bilinear_skycoord(coord, sm.stokes.value[0,0,:]))
+  6540.375582405899
+
+  >>> a = (sm.gl.radian/math.pi)[:3]
+  >>> z = np.cos(sm.gb.radian)[:3]
+  >>> fig, ax = plt.subplots() # doctest: +SKIP
+  >>> ax.scatter(a, z, alpha = 0) # doctest: +SKIP
+  >>> ax.set_xlabel("phi / pi") # doctest: +SKIP
+  >>> ax.set_ylabel("z = cos(theta)") # doctest: +SKIP
+  >>> for i, txt in enumerate(sm.hpx_inds[:3]): # doctest: +SKIP
+  ...     #adds pixel index at center of each pixel
+  ...     ax.annotate(txt, (a[i], z[i]), fontsize=8) # doctest: +SKIP
+  >>> for j in range(len(sm.hpx_inds[:3])): # doctest: +SKIP
+  ...     lon = hp.boundaries_lonlat(sm.hpx_inds[j], 100)[0]/math.pi # doctest: +SKIP
+  ...     lat = np.cos(hp.boundaries_lonlat(sm.hpx_inds[j], 100)[1]) # doctest: +SKIP
+  ...     lon = lon.value # doctest: +SKIP
+  ...     lat = lat.value # doctest: +SKIP
+  ...     vertices = np.vstack([lon.ravel(), lat.ravel()]).transpose() # doctest: +SKIP
+  ...     p = Polygon(vertices, closed=True, edgecolor="black", facecolor="none") # doctest: +SKIP
+  ...     # adds boundaries around each pixel
+  ...     ax.add_patch(p) # doctest: +SKIP
+
+  >>> a = (sm.gl.radian/math.pi)[:3]
+  >>> z = np.cos(sm.gb.radian)[:3]
+  >>> fig, ax = plt.subplots() # doctest: +SKIP
+  >>> ax.scatter(a, z, alpha = 0) # doctest: +SKIP
+  >>> ax.set_xlabel("phi / pi") # doctest: +SKIP
+  >>> ax.set_ylabel("z = cos(theta)") # doctest: +SKIP
+  >>> # nested instead of ring
+  >>> for i, txt in enumerate(hp.ring_to_nested(sm.hpx_inds)[:3]): # doctest: +SKIP
+  ...     ax.annotate(txt, (a[i], z[i]), fontsize=8) # doctest: +SKIP
+  >>> for j in range(len(sm.hpx_inds[:3])): # doctest: +SKIP
+  ...     lon = hp.boundaries_lonlat(sm.hpx_inds[j], 100)[0]/math.pi # doctest: +SKIP
+  ...     lat = np.cos(hp.boundaries_lonlat(sm.hpx_inds[j], 100)[1]) # doctest: +SKIP
+  ...     lon = lon.value # doctest: +SKIP
+  ...     lat = lat.value # doctest: +SKIP
+  ...     vertices = np.vstack([lon.ravel(), lat.ravel()]).transpose() # doctest: +SKIP
+  ...     p = Polygon(vertices, closed=True, edgecolor="black", facecolor="none") # doctest: +SKIP
+  ...     ax.add_patch(p) # doctest: +SKIP
+
+SkyModel: Creating and writing out catalogs
+-------------------------------------------
+
+a) skyh5
+********
+.. code-block:: python
+
+  >>> import os
+  >>> import numpy as np
+  >>> import matplotlib.pyplot as plt # doctest: +SKIP
+  >>> from pyradiosky import SkyModel
+
+  >>> sm = SkyModel(component_type="healpix", nside=1, hpx_inds=[0,1,2,3], stokes=np.zeros((4,1,4)), spectral_type="flat",
+  ...               hpx_order="ring")
+  >>> print(sm.get_lon_lat())
+  (<Longitude [ 45., 135., 225., 315.] deg>, <Latitude [41.8103149, 41.8103149, 41.8103149, 41.8103149] deg>)
+
+  >>> sm.filename == "zero.skyh5"
```

Comment:
You don't need to do this. The filename parameter is actually ignored on the write. It's filled in on read so that if you have an object you can know what file it came out of. So I'd just drop this line (and the other similar ones).